modules/light-clients/06-solomachine/solomachine_test.go

```diff
@@ -46,6 +46,18 @@ func TestSoloMachineTestSuite(t *testing.T) {
 	suite.Run(t, new(SoloMachineTestSuite))
 }
 
+func (suite *SoloMachineTestSuite) TestConnectionHandshake() {
```

Comment:
when channel handshake tests are added, this func can be renamed to `Setup()` and it can be extended to finish a channel handshake

Reply:
This isn't being used anywhere yet, right?
Where do you think we will use it? And should we mock the counterparty(TRY/CONFIRM) handlers too?

Reply:
Currently it is acting as an integration test for setting up connections. We will use it in integration tests for setting up channels and sending/receiving packets. Currently without the fixes in this pr, main will not pass these tests (solo machine has no integration tests with core IBC and several bugs have historically slipped through because of that)